includes/customers/customer-actions.php

```diff
@@ -11,6 +11,136 @@
  * @since       3.0
  */
 
+/**
+ * Is intended to be used when adding a customer directly with edd_add_customer function.
+ *
+ * Because the edd_add_customer function only directly interacts with the customers DB table, we may
+ * need to do some additional actions like adding items to the customer email addresses table.
+ *
+ * @since 3.0.4
+ *
+ * @param int   $customer_id The customer ID that was added.
+ * @param array $data        The data passed in to add the customer.
+ */
+function edd_process_customer_added( $customer_id, $data ) {
+	// Make sure we add a new primary email address to the email addresses table.
+	edd_add_customer_email_address(
+		array(
+			'customer_id' => $customer_id,
+			'email'       => $data['email'],
+			'type'        => 'primary',
+		)
+	);
+}
+add_action( 'edd_customer_added', 'edd_process_customer_added', 10, 2 );
+
+/**
+ * Is intended to be used when updating a customer directly with edd_add_customer function.
+ *
+ * Because the edd_update_customer function only directly interacts with the customers DB table, we may
+ * need to do some additional actions like managing email addresses.
+ *
+ * @since 3.0.4
+ *
+ * @param int          $customer_id       The customer ID being updated.
+ * @param array        $data              The data passed in to update the customer with.
+ * @param EDD_Customer $prev_customer_obj The customer object, prior to these updates.
+ */
+function edd_process_customer_updated( $customer_id, $data, $prev_customer_obj ) {
+	$customer = edd_get_customer( $customer_id );
+
+	// If the email address changed, set the new one as primary.
+	if ( $prev_customer_obj->email !== $customer->email ) {
+
+		// Our transition hook for the type will handle demoting any other email addresses.
+		edd_add_customer_email_address(
+			array(
+				'customer_id' => $customer->id,
+				'email'       => $customer->email,
+				'type'        => 'primary',
+			)
+		);
+	}
+	// Process a User ID change.
+	if ( intval( $customer->user_id ) !== intval( $prev_customer_obj->user_id ) ) {
+		// Attach the User Email to the customer as well.
+		$user = new WP_User( $customer->user_id );
+		if ( $user instanceof WP_User ) {
+			edd_add_customer_email_address(
```

Comment:
This is working, but when I switch users in the customer profile, the main customer email address in the customers table is not updated. So I'm ending up with a customer object with an email  of `originalemail@edd.local` and then two separate email addresses stored in the email addresses table. the `originalemail@edd.local` is deleted from the addresses table, but it's still present on the customer object.

Reply:
@robincornett ready for another set of tests here.